### PR TITLE
ddl: show create table for a local temporary table

### LINKED ISSUE #25884: ddl: `show create table` for a local temporary table
A subtask of https://github.com/pingcap/tidb/issues/24169.

### PR DESCRIPTION
**Pull request by zhaoxugang**
### What problem does this PR solve?

Issue Number: close #25884 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- show create table for a local temporary table

### CODE REVIEW

**Comment by zhaoxugang** [comment]
@djshow832 PTAL

File: executor/ddl.go [hunk 1]
```diff
@@ -246,7 +246,11 @@ func (e *DDLExec) createSessionTemporaryTable(s *ast.CreateTableStmt) error {
 
 	// AutoID is allocated in mocked..
 	alloc := autoid.NewAllocatorFromTempTblInfo(tbInfo)
-	tbl, err := tables.TableFromMeta([]autoid.Allocator{alloc}, tbInfo)
+	allocs := make([]autoid.Allocator, 0, 1)
+	if alloc != nil {
+		allocs = append(allocs, alloc)
+	}
+	tbl, err := tables.TableFromMeta(allocs, tbInfo)
```
Comment (djshow832) [inline executor/ddl.go:253]: why?

File: executor/show_test.go [hunk 2]
```diff
@@ -1375,4 +1375,13 @@ func (s *testSuite5) TestShowTemporaryTable(c *C) {
 		"  KEY `b` (`b`)\n" +
 		") ENGINE=memory DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ON COMMIT DELETE ROWS"
 	tk.MustQuery("show create table t5").Check(testkit.Rows("t5 " + expect))
+
+	tk.MustExec("set tidb_enable_noop_functions=true")
+	tk.MustExec("create temporary table t6 (i int primary key, j int)")
```
Comment (djshow832) [inline executor/show_test.go:1380]: Try to create another table with auto-id.

File: executor/ddl.go (hunk 1 above)
Comment (zhaoxugang) [reply executor/ddl.go:253]: we can know from memid.go L28 that NewAllocatorFromTempTblInfo will be return nil when there is not PK or COMMON PK and not exists auto_incr_id

File: executor/show_test.go (hunk 2 above)
Comment (zhaoxugang) [reply executor/show_test.go:1380]: done

File: executor/ddl.go (hunk 1 above)
Comment (tiancaiamao) [reply executor/ddl.go:253]: When `alloc` is nil, can the current code work?

Comment (tiancaiamao) [reply executor/ddl.go:253]: I mean, the read and write operation ... if `alloc` is nil ...

**Review by tiancaiamao — APPROVED** [review]

File: executor/show_test.go (hunk 2 above)
Comment (zimulala) [reply executor/show_test.go:1380]: Does it need to check the result of `show create table`?

Comment (zhaoxugang) [reply executor/show_test.go:1380]: > Does it need to check the result of `show create table`?

done

**Comment by zhaoxugang** [comment]
\rebuild

**Review by AilinKid — APPROVED** [review]
/LGTM

**Pull request merged**